docs/active-checks.md

```diff
@@ -0,0 +1,255 @@
+- Logs are available directly in the Job’s Pod.
+- No reactions are applied; only status is updated.
+
+### Slurm Jobs (`slurmJob`)
+- The CronJob spawns a Kubernetes Job that **submits a Slurm batch job**.
+- Batch may contain one or many Slurm jobs.
+- Logs are written under `/mnt/jail/opt/soperator-outputs/slurm_jobs/`.
+- After the run completes, the Jobs Controller may apply success/failure reactions (see [Reactions fields (spec)](#reactions-fields-spec)); affected nodes come from Slurm’s `GetNodeList()`.
+
+#### Slurm job submission modes
+- **Default** — One Slurm batch per run.
+- **eachWorkerJobArray** — Run once per worker using a Slurm job array.
+- **eachWorkerJobs** — Run once per worker using separate Slurm jobs.
+- **maxNumberOfJobs** — Limits concurrent jobs; if less than the number of workers, only a subset executes.
+
+#### Slurm partitions (defaults)
+There are three partitions by default:
+- **main** — used for clients’ jobs.
+- **hidden** — hidden partition with the same priority as `main`.
+- **background** — hidden partition with lower priority than `main` and `hidden`.
+
+Submission modes use partitions as follows:
+- **Default** (single job) → `hidden` partition.
+- **eachWorkerJobs** (separate job per worker) → `hidden` partition.
+- **eachWorkerJobArray** (job array per worker) → `background` partition.
+
+Using `background` for job arrays avoids nodes stuck in **PLANNED** and prevents the `main` partition queue from being blocked by large arrays.
+
+## Observability
+
+The Active Checks framework integrates with the cluster observability stack.
+
+### Logging
+- **K8s jobs** → logs are available from the Pods of the Kubernetes Jobs.
+- **Slurm jobs** → logs are written under `/mnt/jail/opt/soperator-outputs/slurm_jobs/`.
+- Other logs (e.g., passive checks) also exist under the broader `/soperator-outputs/` path, but are out of scope for this doc.
+
+### Dashboards
+Results and metrics are visualized in [Grafana](https://grafana.nebius.dev/d/aevt1f8eixhc0f/gpu-health-checker3a-runs). Operators can quickly see:
+- Recent check runs and their outcomes.
+- Historical success/failure rates.
+- Node-level health across multiple checks.
+
+## Controllers
+
+Active Checks are managed by two controllers. Together they implement a GitOps-friendly flow:
+**ActiveCheck CR → CronJob (1:1) → Jobs → Status & (if Slurm) Reactions**.
+
+### 1. Active Check Controller
+
+**Purpose**
+- Watches `ActiveCheck` CRs and reconciles each CR **into exactly one Kubernetes CronJob**.
+- Encodes CronJob settings from the CR (`schedule`, `suspend`, `activeDeadlineSeconds`, history limits, `runAfterCreation`).
+- Ensures optional script sources are wired (e.g., inline `slurmJobSpec.sbatchScript` via ConfigMap).
+- Adds/removes the `slurm.nebius.ai/activecheck-finalizer` to support safe teardown.
+
+**High-level flow**
+1. On create/update:
+    - Validate dependencies (`dependsOn`) and check Slurm cluster readiness.
```

Comment:
Can we describe that we can run check1 only then check2 if they have dependencies?